### PR TITLE
Added mdl-chip component

### PR DESCRIPTION
**Pull request by b-kelly**
Added mdl-chip component as referenced in the [MD specs](https://www.google.com/design/spec/components/chips.html) and issue #857.

I only added the chips, _not_ any of the open, focused, etc elements. These are considerably more complex and need some js to fuel interactions. I think just the chips covers a large percentage of the desired use cases.

I didn't write any docs other than the demo snippets (not sure if I'm qualified).

This is my first contribution to this project, so let me know if my code quality / style needs work to bring it up to par with existing standards.

### CODE REVIEW

File: src/chip/_chip.scss [hunk 1]
```diff
@@ -0,0 +1,63 @@
+
```
Comment (Garbee) [inline src/chip/_chip.scss]: Needs the project license at the top.

File: src/chip/_chip.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        padding: 0;
+        margin: 0;
+        margin-left: 4px;
+        font-size: $chip-font-size;
+
+        //a
+        text-decoration: none;
+        color: $text-color-primary;
+
+        //button
```
Comment (Garbee) [inline src/chip/_chip.scss]: Drop these inline comments, they aren't useful.

File: src/chip/_chip.scss [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    line-height: $chip-height;
+    padding: 0 12px;
+    border-radius: $chip-height / 2;
+    background-color: $chip-bg-color;
+    display: inline-block;
+    color: $text-color-primary;
+    margin: 2px 0;
+    font-size: 0;
+    
+    &.mdl-chip--deletable {
```
Comment (Garbee) [inline src/chip/_chip.scss]: Let's try to keep modifiers at the bottom of declarations. I've learned in recent experimenting it helps with the specificity flow.

File: src/material-design-lite.scss [hunk 4]
```diff
@@ -51,3 +51,4 @@
 @import "tooltip/tooltip";
 @import "shadow/shadow";
 @import "grid/grid";
+@import "chip/chip";
```
Comment (Garbee) [inline src/material-design-lite.scss]: Components should all be alphabetical *except* for shadow/grid which come last for maximum specificity.

**Comment by Garbee** [comment]
Inline comments on some code style things. I haven't reviewed in comparison with the MD spec yet, so that will come later when I have more time.

Other than that:

Do one of the examples as a button and make sure the styling works as that too. That will let you do active/hover/etc states easily. Then people when using `<span>` or whatever else can use `tabindex` to make it activatable.

Documentation will be needed. Chips are fairly easy though and no JS is required (even when doing buttons for states.) So it should be easy enough to do a rough draft looking at the current docs.

Thanks so much on the effort here. This is a much-requested addition.

**Comment by Garbee** [comment]
OH yea, before addressing the comments you're going to need to close this PR. Then start a new branch from `mdl-1.2` and do the addition there. Semantic Versioning keeps us from adding new things in patch releases, so this needs to go into our next minor.

**Comment by b-kelly** [comment]
Thanks for taking a look at this, much appreciated. The inline comments are very helpful. I'll have the updates/improvements done soon, most likely tomorrow or early next week.

For now I'm closing this PR as requested in favor of the future PR on the 1.2 branch. (My bad on that one)

**Pull request closed without merging**